src/internal/registry.js

```diff
@@ -0,0 +1,41 @@
+const registry = new Map();
+
+/**
+ * Returns the handler for `kind` with `ìd`. If there is no handler associated
+ * with `kind` and `id`, returns `undefined`.
+ */
+export function getHandler(kind, id) {
+  const handlers = registry.get(kind);
+  if (handlers) {
+    return handlers.get(id);
+  }
+}
+
+/**
+ * Registers a handler for `kind` with `id`. Previously registered handlers with
+ * the same `kind` and `id` will be overwritten.
+ */
+export function registerHandler(kind, id, handlerFn) {
+  if (getHandler(kind, id)) {
+    console.warn("overwriting", kind, "handler for", id);
+  }
+  if (!registry.has(kind)) {
+    registry.set(kind, new Map());
+  }
+  registry.get(kind).set(id, handlerFn);
+  return handlerFn;
+}
+
+/**
+ * Clears handlers from the registry. If `id` is undefined clears all handlers
+ * for `kind`. If `id` and `kind` is undefined, clears all handlers.
+ */
+export function clearHandlers(kind, id) {
+  if (kind === undefined && id === undefined) {
+    registry.clear();
+  } else if (id === undefined) {
+    registry.delete(kind);
+  } else {
+    registry.get(kind).delete(id);
```

Comment:
Is this call always safe, with regards to the chained call into the map for `kind`?

Reply:
You're right, it's not... I'm still wondering if it's a good idea to have this *global* registry where connectors register handlers by kind. Maybe it's better to create a registry per kind and use this internally in the corresponding connector module. What do you think?

Reply:
I think it's quite alright at the moment. There's of course cohesion to gain from a move, but I don't think it's that important right now.

_First make it work._

Reply:
I simplified this by completely removing the global registry. Will re-add it as soon as there is a real use case.